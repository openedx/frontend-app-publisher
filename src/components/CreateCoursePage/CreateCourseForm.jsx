import moment from 'moment';
import React from 'react';
import { Field, reduxForm } from 'redux-form';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import ReduxFormCreatableSelect from '../ReduxFormCreatableSelect';
import RenderInputTextField from '../RenderInputTextField';
import RenderSelectField from '../RenderSelectField';
import ActionButton from '../ActionButton';
import ButtonToolbar from '../ButtonToolbar';
import FieldLabel from '../FieldLabel';
import PriceList from '../PriceList';

import {
  DATE_INPUT_PATTERN,
} from '../../data/constants';
import {
  endDateHelp, runTypeHelp, pacingHelp, startDateHelp, titleHelp, typeHelp, keyHelp,
} from '../../helpText';
import DateTimeField from '../DateTimeField';
import {
  isSafari, localTimeZone, getDateWithDashes, getOptionsData, parseCourseTypeOptions, parseOptions,
} from '../../utils';
import { basicValidate, handleStafferOrCreateFormFail } from '../../utils/validation';

import './CreateCourseForm.scss';

class BaseCreateCourseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { canSetRunKey: false };
  }

  componentDidUpdate(prevProps) {
    const {
      change,
      courseOptions,
      currentFormValues: {
        type: currentType,
        org: currentOrg,
      },
      organizations,
    } = this.props;
    const {
      currentFormValues: {
        type: prevType,
        org: previousOrg,
      },
    } = prevProps;
    if (currentType !== prevType) {
      const courseOptionsData = getOptionsData(courseOptions);
      const parsedTypeOptions = courseOptionsData
        && parseCourseTypeOptions(courseOptionsData.type.type_options);
      const { courseRunTypeOptions } = parsedTypeOptions;
      change('run_type', courseRunTypeOptions[currentType][1].value);
    }
    if (currentOrg !== previousOrg) {
      const selectedOrganization = organizations.find(org => org.key === currentOrg);
      if (selectedOrganization) {
        this.setRunKeyState(selectedOrganization);
      }
    }
  }

  setRunKeyState(selectedOrganization) {
    this.setState({ canSetRunKey: !selectedOrganization.auto_generate_course_run_keys });
  }

  processOrganizations(organizations) {
    let orgSelectList = [];
    if (organizations) {
      const newOrgs = organizations.map(org => (
        { label: org.name, value: org.key, autoGenerateKey: org.auto_generate_course_run_keys }
      ));
      newOrgs.sort((a, b) => a.label.localeCompare(b.label, undefined, { sensitivity: 'base' }));
      orgSelectList = orgSelectList.concat(newOrgs);
    }
    return orgSelectList;
  }

  // TODO: Removing this for now, as we hide the source field for now and it's not being used.
  //   processSources(sources) {
  //     let sourceSelectList = [{ label: 'Select product source', value: '' }];
  //     if (sources) {
  //       const newSources = sources.map(source => (
  //         { label: source.name, value: source.slug }
  //       ));
  //       sourceSelectList = sourceSelectList.concat(newSources);
  //     }
  //     return sourceSelectList;
  //   }

  render() {
    const {
      currentFormValues,
      handleSubmit,
      organizations,
      pristine,
      isCreating,
      courseOptions,
      courseRunOptions,
    } = this.props;
    const { canSetRunKey } = this.state;
    const courseOptionsData = getOptionsData(courseOptions);
    const parsedTypeOptions = courseOptionsData
      && parseCourseTypeOptions(courseOptionsData.type.type_options);
    const { courseTypeOptions } = parsedTypeOptions;
    const { courseRunTypeOptions } = parsedTypeOptions;
    const { priceLabels } = parsedTypeOptions;
    const courseRunOptionsData = getOptionsData(courseRunOptions);
    const { pacing_type: { choices } } = courseRunOptionsData;
    const pacingTypeOptions = courseRunOptionsData && parseOptions(choices);

    return (
      <div className="create-course-form">
        <h2>Create New Course</h2>
        <hr />
        <form onSubmit={handleSubmit}>
          <Field
            name="org"
            component={ReduxFormCreatableSelect}
            label={<FieldLabel text="Organization" />}
            options={this.processOrganizations(organizations)}
            validate={basicValidate}
            placeholder="Organization..."
          />
          {/* TODO: Removing this field because publisher is currently only used for creating Edx courses.
            We will unhide this field once we start supporting other products as well in publisher. */}
          {/* <Field
            name="source"
            component={RenderSelectField}
            options={this.processSources(sources)}
            label={<FieldLabel text="Source" required />}
            defaultValue={DEFAULT_PRODUCT_SOURCE}
            hidden
            required
          /> */}
          <Field
            name="title"
            component={RenderInputTextField}
            type="text"
            label={(
              <FieldLabel
                id="title-label"
                text="Title"
                required
                helpText={titleHelp}
              />
            )}
            required
          />
          <Field
            name="number"
            component={RenderInputTextField}
            type="text"
            label={(
              <FieldLabel
                id="number-label"
                text="Number"
                required
                extraText="Cannot edit after submission"
                helpText={(
                  <div>
                    <p>
                      Maximum 10 characters. Characters can be letters, numbers, periods,
                      underscores, or hyphens.
                    </p>
                    <p>
                      If a course consists of several modules, the course number can have an
                      ending such as .1x or .2x.
                    </p>
                    <p>
                      <a
                        href="https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/set_up_course/planning_course_information/title_number_guidelines.html#course-number-guidelines"
                        target="_blank"
                        rel="noopener noreferrer"
                      >
                        Learn more.
                      </a>
                    </p>
                    <p><b>Examples:</b></p>
                    <ol>
                      <li>CS002x</li>
                      <li>BIO1.1x, BIO1.2x</li>
                    </ol>
                  </div>
                )}
              />
            )}
            required
          />
          <Field
            name="type"
            component={RenderSelectField}
            options={courseTypeOptions}
            label={(
              <FieldLabel
                id="course-type-label"
                text="Course enrollment track"
                required
                helpText={typeHelp}
              />
            )}
            required
          />
          <PriceList
            priceLabels={currentFormValues.type ? priceLabels[currentFormValues.type] : {}}
            required
          />
          <h2>First run of your Course</h2>
          <hr />
          {canSetRunKey
            && (
            <Field
              name="courseRunKey"
              component={RenderInputTextField}
              type="text"
              pattern="[a-zA-Z0-9-]+"
              label={(
                <FieldLabel
                  id="courseRunKey-label"
                  text="Run Key"
                  helpText={keyHelp}
                  optional
                />
              )}
            />
            )}
          {/* TODO this should be refactored when paragon supports safari */}
          {/* text inputs for safari */}
          {isSafari
            ? (
              <div>
                <Field
                  name="start"
                  type="text"
                  component={DateTimeField}
                  dateLabel="Start date"
                  timeLabel={`Start time (${localTimeZone})`}
                  helpText={startDateHelp}
                  required
                  maxLength="10"
                  pattern={DATE_INPUT_PATTERN}
                  placeholder="yyyy/mm/dd"
                />
                <Field
                  name="end"
                  type="text"
                  component={DateTimeField}
                  dateLabel="End date"
                  timeLabel={`End time (${localTimeZone})`}
                  helpText={endDateHelp}
                  required
                  maxLength="10"
                  pattern={DATE_INPUT_PATTERN}
                  placeholder="yyyy/mm/dd"
                />
              </div>
            )
            // date inputs for all browsers besides safari
            : (
              <div>
                <Field
                  name="start"
                  type="date"
                  component={DateTimeField}
                  dateLabel="Start date"
                  timeLabel={`Start time (${localTimeZone})`}
                  helpText={startDateHelp}
                  required
                  minDate={getDateWithDashes(moment())}
                />
                <Field
                  name="end"
                  type="date"
                  component={DateTimeField}
                  dateLabel="End date"
                  timeLabel={`End time (${localTimeZone})`}
                  helpText={endDateHelp}
                  required
                  minDate={getDateWithDashes(moment(currentFormValues.start).add(1, 'd') || moment())}
                />
              </div>
            )}
          <Field
            name="run_type"
            component={RenderSelectField}
            options={currentFormValues.type ? courseRunTypeOptions[currentFormValues.type] : [{ label: 'Select Course enrollment track first', value: '' }]}
            label={(
              <FieldLabel
                id="course-run-type-label"
                text="Course run enrollment track"
                required
                helpText={runTypeHelp}
              />
            )}
            required
          />
          <Field
            name="pacing_type"
            type="text"
            component={RenderSelectField}
            options={pacingTypeOptions}
            label={(
              <FieldLabel
                id="pacing_type.label"
                text="Course pacing"
                helpText={pacingHelp}
              />
            )}
          />
          <ButtonToolbar>
            <Link to="/">
              <button
                type="button"
                className="btn btn-outline-primary"
                disabled={isCreating}
              >
                Cancel
              </button>
            </Link>
            <ActionButton
              disabled={pristine}
              labels={{
                default: 'Create',
                pending: 'Creating',
              }}
              state={isCreating ? 'pending' : 'default'}
            />
          </ButtonToolbar>
        </form>
      </div>
    );
  }
}

BaseCreateCourseForm.defaultProps = {
  change: () => null,
  isCreating: false,
  pristine: true,
  currentFormValues: {},
};

BaseCreateCourseForm.propTypes = {
  change: PropTypes.func,
  handleSubmit: PropTypes.func.isRequired,
  initialValues: PropTypes.shape({ // eslint-disable-line react/no-unused-prop-types
    org: PropTypes.string,
    source: PropTypes.string,
    title: PropTypes.string,
    number: PropTypes.string,
    type: PropTypes.string,
    prices: PropTypes.shape(),
    start: PropTypes.string,
    end: PropTypes.string,
    upgrade_deadline_override: PropTypes.string,
  }).isRequired,
  currentFormValues: PropTypes.shape({
    org: PropTypes.string,
    type: PropTypes.string,
    start: PropTypes.string,
  }),
  organizations: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    key: PropTypes.string.isRequired,
  })).isRequired,
  pristine: PropTypes.bool,
  isCreating: PropTypes.bool,
  courseOptions: PropTypes.shape({
    data: PropTypes.shape(),
    error: PropTypes.arrayOf(PropTypes.string),
    isFetching: PropTypes.bool,
  }).isRequired,
  courseRunOptions: PropTypes.shape({
    data: PropTypes.shape(),
    error: PropTypes.arrayOf(PropTypes.string),
    isFetching: PropTypes.bool,
  }).isRequired,
};

export default reduxForm({
  form: 'create-course-form',
  onSubmitFail: handleStafferOrCreateFormFail,
})(BaseCreateCourseForm);
export { BaseCreateCourseForm };
