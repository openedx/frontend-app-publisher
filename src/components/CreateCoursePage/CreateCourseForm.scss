.create-course-form {
    // react-select specific styles
    .react-select__control {
      cursor: text;
      border-color: #707070;
      // match the height of edx-paragon form fields
      height: calc(1.3333em + 1.125rem + 2px);
    }
    
    .react-select__dropdown-indicator{
        color: #707070;
    }
  
    .react-select__control--menu-is-open{
      .react-select__single-value{
        color: #D3D3D3
      }
    }
  }
