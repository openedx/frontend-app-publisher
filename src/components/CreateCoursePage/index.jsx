import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';

import CreateCourseForm from './CreateCourseForm';
import LoadingSpinner from '../LoadingSpinner';
import PageContainer from '../PageContainer';
import StatusAlert from '../StatusAlert';
import ConfirmationModal from '../ConfirmationModal';

import { formatPriceData } from '../../utils';

class CreateCoursePage extends React.Component {
  constructor(props) {
    super(props);
    this.handleCourseCreate = this.handleCourseCreate.bind(this);
    this.showModal = this.showModal.bind(this);
    this.cancelCreate = this.cancelCreate.bind(this);
    this.continueCreate = this.continueCreate.bind(this);
    this.state = {
      startedFetching: false,
      createConfirmVisible: false,
      createCourseData: {},
    };

    if (props.courseInfo.error) {
      props.clearCourseInfoErrors();
    }
  }

  componentDidMount() {
    this.props.fetchOrganizations();
    this.props.fetchCourseOptions();
    this.props.fetchCourseRunOptions();
    this.setStartedFetching();
  }

  setStartedFetching() {
    this.setState({ startedFetching: true });
  }

  handleCourseCreate(options) {
    const priceData = formatPriceData(options, this.props.courseOptions);
    const courseData = {
      ...priceData,
      org: options.org,
      title: options.title,
      number: options.number,
      type: options.type,
      course_run: {
        ...priceData, // might be sending more than we need to, but :shrug:
        start: options.start,
        end: options.end,
        pacing_type: options.pacing_type,
        run_type: options.run_type,
        term: options.courseRunKey,
      },
    };
    return this.props.createCourse(courseData);
  }

  showModal(options) {
    this.setState({
      createCourseData: options,
      createConfirmVisible: true,
    });
  }

  continueCreate() {
    const {
      createCourseData,
    } = this.state;

    this.setState({
      createCourseData: {},
      createConfirmVisible: false,
    });

    this.handleCourseCreate(createCourseData);
  }

  cancelCreate() {
    const {
      clearCreateCourseStatus,
    } = this.props;

    this.setState({
      createCourseData: {},
      createConfirmVisible: false,
    });
    return clearCreateCourseStatus();
  }

  render() {
    if (!this.props.publisherUserInfo) {
      return (
        <StatusAlert
          id="error"
          alertType="danger"
          title="Course Create Form failed to load: "
          message="User information unavailable"
        />
      );
    }

    const {
      initialValues,
      courseInfo,
      publisherUserInfo,
      formValues,
      courseOptions,
      courseRunOptions,
    } = this.props;
    const {
      startedFetching,
      createConfirmVisible,
    } = this.state;

    const organizations = publisherUserInfo.organizations ? publisherUserInfo.organizations : [];
    if (organizations.length === 1) { initialValues.org = organizations[0].key; }

    const errorArray = [];
    if (courseInfo.error) {
      courseInfo.error.forEach((error, index) => {
        errorArray.push(error);
        if (index < courseInfo.error.length) {
          errorArray.push(<br />);
        }
      });
    }

    if (publisherUserInfo.error) {
      publisherUserInfo.error.forEach((error, index) => {
        errorArray.push(error);
        if (index < publisherUserInfo.error.length) {
          errorArray.push(<br />);
        }
      });
    }

    const showSpinner = !startedFetching || publisherUserInfo.isFetching
      || courseOptions.isFetching || courseRunOptions.isFetching;
    const showForm = startedFetching && !publisherUserInfo.isFetching
      && !courseOptions.isFetching && !courseRunOptions.isFetching;

    return (
      <>
        <Helmet>
          <title>Create a New Course</title>
        </Helmet>

        <ConfirmationModal
          title="Create a New Course?"
          body="This will create a new course in studio. Confirm that your course number is correct, as it cannot be changed later."
          buttonLabel="Create"
          open={createConfirmVisible}
          onSubmit={this.continueCreate}
          onClose={this.cancelCreate}
        />

        <PageContainer>
          { showSpinner && <LoadingSpinner /> }
          { showForm
          && (
            <div>
              <CreateCourseForm
                id="create-course-form"
                onSubmit={this.showModal}
                initialValues={initialValues}
                currentFormValues={formValues}
                organizations={organizations}
                isCreating={courseInfo.isCreating}
                courseOptions={courseOptions}
                courseRunOptions={courseRunOptions}
              />
              {errorArray.length > 1 && (
                <StatusAlert
                  id="create-error"
                  alertType="danger"
                  message={errorArray}
                />
              ) }
            </div>
          )}
        </PageContainer>
      </>
    );
  }
}

CreateCoursePage.defaultProps = {
  initialValues: {},
  publisherUserInfo: {
    organizations: [],
  },
  fetchOrganizations: () => {},
  fetchCourseOptions: () => {},
  fetchCourseRunOptions: () => {},
  courseOptions: {},
  courseRunOptions: {},
  courseInfo: {},
  createCourse: () => {},
  formValues: {},
  clearCourseInfoErrors: () => null,
  clearCreateCourseStatus: () => {},
};

CreateCoursePage.propTypes = {
  initialValues: PropTypes.shape({ // eslint-disable-line react/no-unused-prop-types
    org: PropTypes.string,
    title: PropTypes.string,
    number: PropTypes.string,
    type: PropTypes.string,
    prices: PropTypes.shape(),
    start: PropTypes.string,
    end: PropTypes.string,
    pacing_type: PropTypes.string,
  }),
  publisherUserInfo: PropTypes.shape({
    organizations: PropTypes.array,
    error: PropTypes.arrayOf(PropTypes.string),
    isFetching: PropTypes.bool,
  }),
  courseInfo: PropTypes.shape({
    error: PropTypes.arrayOf(PropTypes.string),
    data: PropTypes.shape({
      uuid: PropTypes.string,
    }),
    isCreating: PropTypes.bool,
  }),
  fetchOrganizations: PropTypes.func,
  fetchCourseOptions: PropTypes.func,
  fetchCourseRunOptions: PropTypes.func,
  courseOptions: PropTypes.shape({
    data: PropTypes.shape(),
    error: PropTypes.arrayOf(PropTypes.string),
    isFetching: PropTypes.bool,
  }),
  courseRunOptions: PropTypes.shape({
    data: PropTypes.shape(),
    error: PropTypes.arrayOf(PropTypes.string),
    isFetching: PropTypes.bool,
  }),
  createCourse: PropTypes.func,
  formValues: PropTypes.shape({}),
  clearCourseInfoErrors: PropTypes.func,
  clearCreateCourseStatus: PropTypes.func,
};

export default CreateCoursePage;
