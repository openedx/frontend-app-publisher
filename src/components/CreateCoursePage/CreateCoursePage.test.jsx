import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CreateCoursePage from './index';

const organizations = [{ name: 'edX', key: 'edx' }, { name: 'edX2', key: 'edx2' }];
const sources = [{ name: 'souce1', slug: 'source1' }, { name: 'source2', slug: 'source2' }];

describe('CreateCoursePage', () => {
  it('renders html correctly', () => {
    const component = shallow(<CreateCoursePage />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders page correctly while fetching', () => {
    const component = shallow(<CreateCoursePage
      fetchOrganizations={() => null}
      publisherUserInfo={{
        organizations: [],
        error: null,
        isFetching: true,
      }}
      productSourceOptions={{
        productSources: [],
        error: null,
        isFetching: true,
      }}
      courseInfo={{
        error: null,
        isCreating: false,
        data: {},
      }}
      createCourse={() => null}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders page correctly with organizations', () => {
    const component = shallow(<CreateCoursePage
      fetchOrganizations={() => null}
      publisherUserInfo={{
        organizations,
        error: null,
        isFetching: false,
      }}
      productSourceOptions={{
        productSources: sources,
        error: null,
        isFetching: false,
      }}
      courseInfo={{
        error: null,
        isCreating: false,
        data: {},
      }}
      createCourse={() => null}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders page correctly with org error', () => {
    const component = shallow(<CreateCoursePage
      fetchOrganizations={() => null}
      publisherUserInfo={{
        organizations,
        error: ['Fail'],
        isFetching: false,
      }}
      courseInfo={{
        error: null,
        isCreating: false,
        data: {},
      }}
      createCourse={() => null}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders page correctly with no publisherUserInfo', () => {
    const component = shallow(<CreateCoursePage
      fetchOrganizations={() => null}
      courseInfo={{
        error: null,
        isCreating: false,
        data: {},
      }}
      createCourse={() => null}
      publisherUserInfo={null}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders page correctly with course create error', () => {
    const component = shallow(<CreateCoursePage
      fetchOrganizations={() => null}
      publisherUserInfo={{
        organizations,
        error: null,
        isFetching: false,
      }}
      courseInfo={{
        error: ['Fail'],
        isCreating: false,
        data: {},
      }}
      createCourse={() => null}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders page correctly with course create success', () => {
    const component = shallow(<CreateCoursePage
      fetchOrganizations={() => null}
      publisherUserInfo={{
        organizations,
        error: null,
        isFetching: false,
      }}
      courseInfo={{
        error: null,
        isCreating: false,
        data: {
          uuid: '11111111-1111-1111-1111-111111111111',
        },
      }}
      createCourse={() => null}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('renders page correctly with course create in progress', () => {
    const component = shallow(<CreateCoursePage
      fetchOrganizations={() => null}
      publisherUserInfo={{
        organizations,
        error: null,
        isFetching: false,
      }}
      courseInfo={{
        error: null,
        isCreating: true,
        data: {
          uuid: '11111111-1111-1111-1111-111111111111',
        },
      }}
      createCourse={() => null}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
