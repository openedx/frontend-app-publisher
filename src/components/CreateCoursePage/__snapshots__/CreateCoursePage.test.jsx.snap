// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCoursePage renders html correctly 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create a New Course
    </title>
  </HelmetWrapper>
  <ConfirmationModal
    body="This will create a new course in studio. Confirm that your course number is correct, as it cannot be changed later."
    buttonLabel="Create"
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title="Create a New Course?"
  />
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <ReduxForm
        courseOptions={Object {}}
        courseRunOptions={Object {}}
        currentFormValues={Object {}}
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="create-course-form"
        getFormState={[Function]}
        id="create-course-form"
        initialValues={Object {}}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        organizations={Array []}
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`CreateCoursePage renders page correctly while fetching 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create a New Course
    </title>
  </HelmetWrapper>
  <ConfirmationModal
    body="This will create a new course in studio. Confirm that your course number is correct, as it cannot be changed later."
    buttonLabel="Create"
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title="Create a New Course?"
  />
  <PageContainer
    className=""
    wide={false}
  >
    <LoadingSpinner
      message="Loadingâ€¦"
    />
  </PageContainer>
</Fragment>
`;

exports[`CreateCoursePage renders page correctly with course create error 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create a New Course
    </title>
  </HelmetWrapper>
  <ConfirmationModal
    body="This will create a new course in studio. Confirm that your course number is correct, as it cannot be changed later."
    buttonLabel="Create"
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title="Create a New Course?"
  />
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <ReduxForm
        courseOptions={Object {}}
        courseRunOptions={Object {}}
        currentFormValues={Object {}}
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="create-course-form"
        getFormState={[Function]}
        id="create-course-form"
        initialValues={Object {}}
        isCreating={false}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        organizations={
          Array [
            Object {
              "key": "edx",
              "name": "edX",
            },
            Object {
              "key": "edx2",
              "name": "edX2",
            },
          ]
        }
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
      <StatusAlert
        alertType="danger"
        className=""
        dismissible={false}
        iconClassNames={Array []}
        id="create-error"
        message={
          Array [
            "Fail",
            <br />,
          ]
        }
        onClose={[Function]}
        title={null}
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`CreateCoursePage renders page correctly with course create in progress 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create a New Course
    </title>
  </HelmetWrapper>
  <ConfirmationModal
    body="This will create a new course in studio. Confirm that your course number is correct, as it cannot be changed later."
    buttonLabel="Create"
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title="Create a New Course?"
  />
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <ReduxForm
        courseOptions={Object {}}
        courseRunOptions={Object {}}
        currentFormValues={Object {}}
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="create-course-form"
        getFormState={[Function]}
        id="create-course-form"
        initialValues={Object {}}
        isCreating={true}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        organizations={
          Array [
            Object {
              "key": "edx",
              "name": "edX",
            },
            Object {
              "key": "edx2",
              "name": "edX2",
            },
          ]
        }
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`CreateCoursePage renders page correctly with course create success 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create a New Course
    </title>
  </HelmetWrapper>
  <ConfirmationModal
    body="This will create a new course in studio. Confirm that your course number is correct, as it cannot be changed later."
    buttonLabel="Create"
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title="Create a New Course?"
  />
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <ReduxForm
        courseOptions={Object {}}
        courseRunOptions={Object {}}
        currentFormValues={Object {}}
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="create-course-form"
        getFormState={[Function]}
        id="create-course-form"
        initialValues={Object {}}
        isCreating={false}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        organizations={
          Array [
            Object {
              "key": "edx",
              "name": "edX",
            },
            Object {
              "key": "edx2",
              "name": "edX2",
            },
          ]
        }
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`CreateCoursePage renders page correctly with no publisherUserInfo 1`] = `
<StatusAlert
  alertType="danger"
  className=""
  dismissible={false}
  iconClassNames={Array []}
  id="error"
  message="User information unavailable"
  onClose={[Function]}
  title="Course Create Form failed to load: "
/>
`;

exports[`CreateCoursePage renders page correctly with org error 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create a New Course
    </title>
  </HelmetWrapper>
  <ConfirmationModal
    body="This will create a new course in studio. Confirm that your course number is correct, as it cannot be changed later."
    buttonLabel="Create"
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title="Create a New Course?"
  />
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <ReduxForm
        courseOptions={Object {}}
        courseRunOptions={Object {}}
        currentFormValues={Object {}}
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="create-course-form"
        getFormState={[Function]}
        id="create-course-form"
        initialValues={Object {}}
        isCreating={false}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        organizations={
          Array [
            Object {
              "key": "edx",
              "name": "edX",
            },
            Object {
              "key": "edx2",
              "name": "edX2",
            },
          ]
        }
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
      <StatusAlert
        alertType="danger"
        className=""
        dismissible={false}
        iconClassNames={Array []}
        id="create-error"
        message={
          Array [
            "Fail",
            <br />,
          ]
        }
        onClose={[Function]}
        title={null}
      />
    </div>
  </PageContainer>
</Fragment>
`;

exports[`CreateCoursePage renders page correctly with organizations 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Create a New Course
    </title>
  </HelmetWrapper>
  <ConfirmationModal
    body="This will create a new course in studio. Confirm that your course number is correct, as it cannot be changed later."
    buttonLabel="Create"
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title="Create a New Course?"
  />
  <PageContainer
    className=""
    wide={false}
  >
    <div>
      <ReduxForm
        courseOptions={Object {}}
        courseRunOptions={Object {}}
        currentFormValues={Object {}}
        destroyOnUnmount={true}
        enableReinitialize={false}
        forceUnregisterOnUnmount={false}
        form="create-course-form"
        getFormState={[Function]}
        id="create-course-form"
        initialValues={Object {}}
        isCreating={false}
        keepDirtyOnReinitialize={false}
        onSubmit={[Function]}
        organizations={
          Array [
            Object {
              "key": "edx",
              "name": "edX",
            },
            Object {
              "key": "edx2",
              "name": "edX2",
            },
          ]
        }
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
    </div>
  </PageContainer>
</Fragment>
`;
